reactos/drivers/usb/usbhub/fdo.c

```diff
@@ -1112,7 +1110,10 @@ DestroyUsbChildDeviceObject(
         return STATUS_UNSUCCESSFUL;
     }
 
+    DPRINT("Removing device on port %d (Child index: %d)\n", PortId, Index);
+
     /* Remove the device from the table */
+    // is lock needed?
```

Comment:
In case of toaster, it manages it's FDO list from PDO too (I haven't reviewed their fdo), i.e. PDO is allowed to remove itself from FDO's child list. I our case, I am trying to manage FDO's child list ONLY from DeviceStatusChangeThread() which is called in on connect/disconnect events coming from lower levels. In this way in parent FDO we'll have list of child which is currently presented on the bus.